crates/cast/bin/cmd/wallet/mod.rs

```diff
@@ -320,40 +345,123 @@ flag to set your key via:
                 );
                 println!("{}", Paint::green(success_message));
             }
-            WalletSubcommands::List => {
-                let default_keystore_dir = Config::foundry_keystores_dir()
-                    .ok_or_else(|| eyre::eyre!("Could not find the default keystore directory."))?;
-                // Create the keystore directory if it doesn't exist
-                fs::create_dir_all(&default_keystore_dir)?;
-                // List all files in keystore directory
-                let keystore_files: Result<Vec<_>, eyre::Report> =
-                    std::fs::read_dir(&default_keystore_dir)
-                        .wrap_err("Failed to read the directory")?
-                        .filter_map(|entry| match entry {
-                            Ok(entry) => {
-                                let path = entry.path();
-                                if path.is_file() && path.extension().is_none() {
-                                    Some(Ok(path))
-                                } else {
-                                    None
+            WalletSubcommands::List { dir, ledger, trezor, aws, all } => {
```

Comment:
I think this would be useful in the wallet crate @klkvr 

like a struct that looks up all configured sources and returns the results

so I think we want to do this a bit differently,
the helper struct should gather all the info and then we print them here in the command

Reply:
makes sense, happy to accommodate such after things moved to new crate or leave up to @klkvr lmk what you guys decide

Reply:
thanks, checked and makes sense (I might need to derive builder for `MultiWalletOpts`), will prepare changes and make a PR once #7141 in master

Reply:
I redid PR with new wallet structure, pretty tidy now imo, please review